### ISSUE TITLE
 listener.ssl.external.peer_cert_as_username does not work with an empty username EMQx 4.3.8 and 4.3.9 and MQTTx 

### ISSUE THREAD

**Issue by gunnarpieter** [emqx/emqx#6885]
### Subject of the issue
 listener.ssl.external.peer_cert_as_username does not work with an empty username EMQx 4.3.8 and 4.3.9 and MQTTx 

### Your environment
Docker, self signed certs with common name "client"

For EMQ X 4.3 or newer, please provide the log archive generated by [node_dump tool](https://github.com/emqx/emqx/blob/master/bin/node_dump) 
Not sure how to easily do this on the docker container.

- EMQ X version (e.g. `emqx_ctl status`): 4.3.8
- Hardware configuration (e.g. `lscpu`): docker
- OS (e.g. `cat /etc/os-release`): Alpine Linux
- Kernel (e.g. `uname -a`): Linux 997a7b9f72f3 5.10.16.3-microsoft-standard-WSL2

### Steps to reproduce
Broker and client with two way TLS (https://www.emqx.com/en/blog/enable-two-way-ssl-for-emqx)
Start the broker with allow_anonymous = false.
Have mnesia with username client and password client.
Try to sign in with no username and password client.

### Expected behaviour
Expected to replace the empty username of the client with the cn name within the cert.

### Actual behaviour
MQTTx keeps loading without result. While it works if you enter 1 or more characters.

Missing log file can delay the handling of the issue.

**Comment by savonarola** [comment]
Hello!

I reproduced this issue, and it seems to be an MQTTx problem — when username field is empty, MQTTx opens connection but does not send MQTT connection packet.
This behaviour also can be reproduced for ordinary TCP (nonsecure) connections.

Also I managed to successfully reproduce the desired behaviour.
* I followed https://www.emqx.com/en/blog/enable-two-way-ssl-for-emqx instructions, creating client certificate with `client_uname` CN.
* Set `listener.ssl.external.peer_cert_as_username = on`
* Enabled mnesia auth plugin: `emqx_ctl plugins load emqx_auth_mnesia`.
* Created user: `emqx_ctl user add client_uname pass`.
* Successfully published message with correct password, CN and with any username, empty username or without username at all:
```bash
emqtt pub -P pass -p 8883 -t test --payload test --CAfile ca.pem --cert client.pem --key client.key --enable-ssl
emqtt pub -u '' -P pass -p 8883 -t test --payload test --CAfile ca.pem --cert client.pem --key client.key --enable-ssl
emqtt pub -u anyusername -P pass -p 8883 -t test --payload test --CAfile ca.pem --cert client.pem --key client.key --enable-ssl
```
